### ISSUE TITLE
Command 'Flutter: Run Flutter Doctor' resulted in an error (Cannot read property 'message' of undefined)

### ISSUE THREAD

**Issue by LayYouTube** [Dart-Code/Dart-Code#3145]
Command 'Flutter: Run Flutter Doctor' resulted in an error (Cannot read property 'message' of undefined)
when im gonna run Flutter: Run Flutter Doctor i have this problem

**Comment by DanTup** [comment]
If you open the VS Code developer tools (Help -> Toggle Developer Tools) and click on the Console tab, does the error show in there? If so, can it be expanded to show a stack trace? If so, could you copy/paste or attach a screenshot here? Thanks!

**Comment by LayYouTube** [comment]
![Zrzut ekranu (55)](https://user-images.githubusercontent.com/71594301/108246175-6c685a80-7151-11eb-98d1-b43dbf3b9912.png)

**Comment by DanTup** [comment]
@LayYouTube please click on the Console tab shown here:

![108246175-6c685a80-7151-11eb-98d1-b43dbf3b9912](https://user-images.githubusercontent.com/1078012/108248803-370d3e00-714c-11eb-9c5e-98652e9c24ca.png)

**Comment by LayYouTube** [comment]
![Zrzut ekranu (56)](https://user-images.githubusercontent.com/71594301/108255219-41cfcf00-715c-11eb-8a3c-68bdbec871b5.png)

**Comment by LayYouTube** [comment]
@DanTup

**Comment by DanTup** [comment]
@LayYouTube it looks like Flutter is failing to run - in the output window at the bottom it says the output from Flutter was "Unable to find git in your PATH". According the [requirements](https://flutter.dev/docs/get-started/install/windows#system-requirements) on the Flutter site, you need to install Git first.

However, this should be handled better and not throw an error like "cannot read property 'message' of undefined". I tried to reproduce this, but the error did not occur and my output channel is named "flutter (flutter)" whereas yours is "flutter doctor". Can you confirm what version of VS Code and the Dart extension you have? Thanks!

**Comment by DanTup** [comment]
Actually, I was able to reproduce that error - I'll fix that so it's reported better, though to fix your issue you'll need to install Git.

**Comment by DanTup** [comment]
I think the error message shown here is incorrect (possibly a VS Code bug), it should've actually said something like "The command 'flutter.doctor' is not registered".

However, the cause of that was a bug here. Trying to run "flutter doctor" after the Flutter SDK failed to initialise would not work correctly. That's been fixed for the next version, so in the same situation with the next release, the Run Flutter Doctor command will run fine (although in this instance, would also print "Cannot find Git" or similar to the output).

So in summary:

- Installing Git should fix your problem
- The next version of the Dart extension will make this error clearer rather than presenting an unrelated error message
Thanks!

**Comment by LayYouTube** [comment]
tysm
